### PR TITLE
docs: update README with gardener removal

### PR DESCRIPTION
**Pull request by grischperl**
## Description

**Changes proposed in this pull request:**

Gardener dependency has been removed in the following PRs: #3347 and #3353 
This change should also be reflected in the README.

## Related issue(s)

#3312

### CODE REVIEW

File: components/kyma-metrics-collector/README.md [hunk 1]
```diff
@@ -5,10 +5,9 @@ Kyma Metrics Collector (KMC) is a component that scrapes all Kyma clusters to ge
 
 ## Functionality
 The basic flow for KMC is as follows:
-* KMC workers get a list of runtimes from [Kyma Environment Broker (KEB)](https://github.com/kyma-project/kyma-environment-broker/tree/main). 
- * Additionally, the workers get the kubeconfig, secret and shoot from Gardener. 
- * KMC adds the runtimes to a queue to work through them and re-queues a runtime should an error occur. 
- * Information on PVCs, SVCs and Nodes is retrieved from SAP Kyma Runtime (SKR). 
+* KMC workers get a list of runtimes from [Kyma Environment Broker (KEB)](https://github.com/kyma-project/kyma-environment-broker/tree/main).
+ * KMC adds the runtimes to a queue to work through them and re-queues a runtime should an error occur.
```
Comment (NHingerl) [inline components/kyma-metrics-collector/README.md:9]:
```suggestion
 * KMC adds the runtimes to a queue to work through them. If an error occurs, KMC re-queues the affected runtime.
```

File: components/kyma-metrics-collector/README.md [hunk 2]
```diff
@@ -5,10 +5,9 @@ Kyma Metrics Collector (KMC) is a component that scrapes all Kyma clusters to ge
...
+ * Information on PVCs, SVCs and Nodes is retrieved from SAP Kyma Runtime (SKR).
```
Comment (NHingerl) [inline components/kyma-metrics-collector/README.md:10]:
```suggestion
 * Information on PVCs, SVCs and Nodes is retrieved from SAP BTP, Kyma runtime (SKR).
```
official product name

Comment (grischperl) [reply components/kyma-metrics-collector/README.md]: Done

File: components/kyma-metrics-collector/README.md (hunk 1 above)
Comment (grischperl) [reply components/kyma-metrics-collector/README.md]: Done

**Review by marcobebway — APPROVED** [review]

File: components/kyma-metrics-collector/dev/deployment.yaml [hunk 3]
```diff
@@ -30,7 +30,6 @@ spec:
           - --worker-pool-size=5
           - --log-level=debug
           - --listen-addr=8080
-          - --gardener-namespace=garden-kyma-dev
```
Comment (marcobebway) [inline components/kyma-metrics-collector/dev/deployment.yaml:33]: Is this change needed?

Comment (grischperl) [reply components/kyma-metrics-collector/dev/deployment.yaml:33]: Yes, this flag was removed [here](https://github.com/kyma-project/control-plane/pull/3353/files#diff-ca7aca1159f0207308cbd2ac2674ab21e30376b4c39a2b52babe27ea766badd5L29)

**Review by marcobebway — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]
